dockers-install.sh

```diff
@@ -0,0 +1,71 @@
+# Update
+sudo apt update
+
+# Installing ca-certificates
+if (test -d ca-certificates)
+then
+  echo "ca-certificates already installed"
+else
+  echo "Installing ca-certificates"
+  sudo apt install -y ca-certificates
+fi
+
+# Installing curl
+if (which curl)
+then
+  echo "curl already installed"
+else
+  echo "Installing curl"
+  sudo apt install -y curl 
+fi
+
+# Creating a keyrings directory 
+if (test -d /etc/bin/keyrings)
+then
+  echo "creating keyrings"
+else
+  echo "Creating keyrings"
+  sudo install -m 0755 -d /etc/apt/keyrings
+fi
+
+# Adding a gpg keyring
+if (test -d  /etc/bin/keyrings/docker.asc)
+then
+  echo "gpg keyring already exists"
+else
+  echo "Installing gpg keyring"
+  sudo curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
+fi
+ 
+# setting permissions 
+if (test -d /etc/bin/keyrings/docker.asc)
+then
+  echo " gpg keyring already configured"
+else
+  echo "Setting permissions for gpg keyring"
+  sudo chmod a+r /etc/apt/keyrings/docker.asc
+fi
+
+# Intializing repository, and placing copies in .list files
+if (test -S  /etc/os-release && echo "$VERSION_CODENAME")
+then 
+  echo "Repository already initialized"
+else
+  echo "Initializing repo"
+  echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian
+  $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+fi 
+
+# Update
+sudo apt update
```

Comment:
This is a hard one, but you should be checking when/if you need to run apt update. You can check if docker is in the apt-cache and it's the right version by using something like:

`apt-cache show docker` and if not in the apt cache, then do a `sudo apt update`.

Reply:
Thank you sir, I was not sure how to go about that.

Reply:
I will run some test when I get in front of my machine.